src/zkevm_specs/state.py

```diff
@@ -388,23 +391,35 @@ def check_tx_log(row: Row, row_prev: Row):
     if row.tag() == row_prev.tag():
         if tx_id != pre_tx_id:
             assert tx_id == pre_tx_id + 1
-            assert log_id == 0
+            # in RW table, log_id starts with 1 per tx, zero log_id means no any log step executed,
+            # hence no any log entry inserts into rw table.
+            assert log_id == 1
+            # first field_tag is Address when tx changes
+            assert row.field_tag() == TxLogFieldTag.Address
         else:
             # increase log_id when tag changes to Address, make sure tag can only increase(Non-Decreasing),
-            # when log_index stays same,
+            # when log_id stays same,
             if pre_field_tag == U256(TxLogFieldTag.Address):
                 assert (field_tag - pre_field_tag).n > 0
-            # make sure if tag Data appear, data_index can only increase by one when tag stays same.
-            # make sure if tag Topic appear, topic_index in range [0,4),can only increase by one when tag stays same.
+            # index is zero for address
+            if field_tag == U256(TxLogFieldTag.Address):
+                assert index == 0
+
+            # increase log_id when field tag changes to Address within same tx
+            if tag == prev_tag and field_tag == U256(TxLogFieldTag.Address):
+                assert log_id == prev_log_id + 1
+
+            # within same tx, log_id will not change if field_tag != Address
+            if tag == prev_tag and field_tag != U256(TxLogFieldTag.Address):
+                assert log_id == prev_log_id
+            # if tag Data appear, data_index can only increase by one when tag stays same.
```

Comment:
as we talked possible redundant constraint above, I am not willing to add this in state circuit side as evm circuit has been constrained because they use look up from zero index .